src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,187 @@
+// (2)     util = (targetInterestRate - baseRatePerBlock) * WAD / multiplierPerBlock
+//
+// https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L79
+//
+//         util = borrows * WAD / (cash + borrows - reserves);
+// (3)     cash + borrows - reserves = borrows * WAD / util
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CErc20.sol#L14
+interface CErc20Like {
+    function totalBorrows()           external view returns (uint256);
+    function totalReserves()          external view returns (uint256);
+    function interestRateModel()      external view returns (address);
+    function getCash()                external view returns (uint256);
+    function underlying()             external view returns (address);
+}
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/BaseJumpRateModelV2.sol#L10
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateModelLike        public tack;
+    uint256                      public barb; // target Interest Rate Per Block [wad] (0)
+
+    CErc20Like public immutable cDai;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address cDai_) {
+        address rateModel_ = CErc20Like(cDai_).interestRateModel();
+        require(rateModel_ != address(0), "D3MCompoundPlan/invalid-rateModel");
+
+        cDai = CErc20Like(cDai_);
+        tack = InterestRateModelLike(rateModel_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MCompoundPlan/not-authorized");
+        _;
+    }
+
+    // --- Math ---
+    uint256 internal constant WAD = 10 ** 18;
+    function _wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * y) / WAD;
+    }
+    function _wdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * WAD) / y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "barb") {
```

Comment:
Add appropriate boundary here. I think that means that `data < WAD` but Tal will double-check.

Reply:
We can use this value - https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/CTokenInterfaces.sol#L31

Reply:
Done in 0cdf45a.